lib/publiccloud/instance.pm

```diff
@@ -477,4 +481,202 @@ sub network_speed_test() {
     record_info("curl $rmt_host", $self->run_ssh_command(cmd => "curl -w '$write_out' -o /dev/null -v https://$rmt_host/", proceed_on_failure => 1));
 }
 
+=head2 measure_boottime
+
+    measure_boottime();
+
+Perfomrance measurement of the system Boot time. 
+Mainly used C<systemd-analyze> command for the data extraction.
+Data is then collected in an internal record, ready for storing in a DB.
+- when PUBLIC_CLOUD_PERF_COLLECT=0 boottime measurement disabled;
+- when bit-0 ON,First reboot only enabled =1;
+- when bit-1 ON, first,softreboot enabled =3 [2 too];
+- when bit-2 ON, first,hardreboot enabled =5 [4 too];
+- when all bits ON,all 3 measures enabled =7 [6 too].
+
+=cut
+
+sub measure_boottime() {
+    my ($self, $instance) = @_;
+    my $perf = get_var('PUBLIC_CLOUD_PERF_COLLECT');
+    return 0 unless ($perf);
+
+    my $softreboot = $perf & 2;    # bit-1 check
+    my $hardreboot = $perf & 4;    # bit-2 check
+
+    my $ret = {
+        kernel_release => undef,
+        kernel_version => undef,
+        analyze => {},
+        blame => {
+            first => {}, soft => {}, hard => {}
+        },
+    };
+    record_info("BOOT TIME", 'systemd_analyze');
+    # first deployment analysis
+    my ($systemd_analyze, $systemd_blame) = do_systemd_analyze($instance);
+    $ret->{analyze}->{$_} = $systemd_analyze->{$_} foreach (keys(%{$systemd_analyze}));
+    $ret->{blame}->{first} = $systemd_blame;
```

Comment:
```suggestion
    $ret->{blame} = $systemd_blame;
    $ret->{type} = 'first';
```

Reply:
ok.

**Note**: considering also the proposal to restore the original `boottime.md` until removal decision made, after this data_record will be changed here and on _Influxdb_ too, it will be _different_ from the original one.
Therefore `boottime.md` can be used for only _local display_, having care to always set  `PUBLIC_CLOUD_PERF_COLLECT` and `_PUBLIC_CLOUD_PERF_PUSH_DATA` to **false**